src/test/java/dk/cngroup/trainings/spring/springassignment/service/AnimalServiceImplTest.java

```diff
@@ -2,242 +2,201 @@
+
+		Assert.assertTrue(animalService.addAnimal(dauphin).isPresent());
+		Assert.assertEquals("Dauphin",
+				animalService.addAnimal(dauphin).get().getName());
+	}
+
+	@Test(expected = InvalidAnimalException.class)
+	public void testAddAnimalWithShortName() throws InvalidAnimalException {
+		// Sad path: Adding an animal with short name, less than 2 characters
+		Animal pigeon = new Animal(6L, "p", "Smartest aqua mammal");
+
+		animalService.addAnimal(pigeon);
+	}
+
+	@Test(expected = InvalidAnimalException.class)
+	public void testAddAnimalWithLongDescription() throws InvalidAnimalException {
+		// Sad path: (Adding an animal with long description, more than 10000)
+		Animal pigeon = new Animal(6L, "p",
+				StringUtils.repeat("*", 10001));
+
+		animalService.addAnimal(pigeon);
+	}
+
+	@Test(expected = InvalidAnimalException.class)
+	public void testAddAnimalWithDescriptionContainingPenguin() throws InvalidAnimalException {
+		// Sad path: Adding an animal with description containing Penguin
+		Animal penguin = new Animal(8L, "Penguin",
+				"Royal Penguins of the north pole");
+
+		animalService.addAnimal(penguin);
+	}
+
+	@Test
+	public void deleteAnimalByExistingId() {
+		//Happy path:
+		long id = 2L;
+		Mockito.when(animalRepositoryMock.existsById(id)).thenReturn(true);
+
+		Assert.assertTrue(animalService.deleteAnimalById(id));
+	}
+
+	@Test
+	public void deleteAnimalByNonExistingId() {
+		//Sad path:
+		long id = 10L; // Non existing id
+		Mockito.when(animalRepositoryMock.existsById(id)).thenReturn(false);
+
+		Assert.assertFalse(animalService.deleteAnimalById(id));
+	}
+
+	@Test
+	public void testUpdateAnimalByExistingId() throws InvalidAnimalException {
+		// Happy path:
+		long id = 3L;
+		Animal oldAnimal = animals.get(2);
+		Animal updatedAnimal = new Animal(100L, "Falcon",
+				"Fastest animal on earth");
+
+		Mockito.when(animalRepositoryMock.existsById(id)).thenReturn(true);
+		Mockito.when(animalRepositoryMock.findById(id))
+				.thenReturn(Optional.ofNullable(oldAnimal));
+		updatedAnimal.setId(oldAnimal.getId());
+		Mockito.when(animalRepositoryMock.save(updatedAnimal))
+				.thenReturn(updatedAnimal);
+
+		updatedAnimal.setId(100L);
+		Optional<Animal> animal =
+				animalService.updateAnimalById(id, updatedAnimal);
+
+		Assert.assertNotNull(animal);
+		Assert.assertEquals("Falcon", animal.get().getName());
+	}
+
+	@Test
+	public void testUpdateAnimalByNonExistingId() throws InvalidAnimalException {
+		// Sad path: Updating an animal with non existing id
+		long id = 10L; // non existing
+		Animal updatedAnimal = new Animal(100L, "Falcon",
+				"Fastest animal on earth");
+		Mockito.when(animalRepositoryMock.existsById(id)).thenReturn(false);
+		Mockito.when(animalRepositoryMock.findById(id))
+				.thenReturn(Optional.empty());
+
+		Optional<Animal> animal2 =
+				animalService.updateAnimalById(id, updatedAnimal);
+
+		Assert.assertFalse(animal2.isPresent());
+	}
+
+    /*
```

Comment:
what is this comment?

Reply:
It means that I don't need to test updating an animal with with an existing id and an invalid animal body, because the update uses addAnimal method, which has been already tested for invalid animal input.